internal/service/alarm_mapper.go

```diff
@@ -0,0 +1,332 @@
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+	neturl "net/url"
+
+	"github.com/imdario/mergo"
+	"github.com/openshift-kni/oran-o2ims/internal/data"
+	"github.com/openshift-kni/oran-o2ims/internal/jq"
+	"github.com/openshift-kni/oran-o2ims/internal/k8s"
+
+	"github.com/itchyny/gojq"
+)
+
+type AlarmMapper struct {
+	logger              *slog.Logger
+	backendClient       *http.Client
+	resourceServerURL   string
+	resourceServerToken string
+	extensions          []string
+	jqTool              *jq.Tool
+}
+
+// AlarmMapperBuilder contains the data and logic needed to create a new AlarmMapper.
+type AlarmMapperBuilder struct {
+	logger        *slog.Logger
+	backendClient *http.Client
+	//transportWrapper    func(http.RoundTripper) http.RoundTripper
+	resourceServerURL   string
+	resourceServerToken string
+	extensions          []string
+}
+
+// AlarmMapper creates a builder that can then be used to configure
+// and create a handler for the AlarmMapper.
+func NewAlarmMapper() *AlarmMapperBuilder {
+	return &AlarmMapperBuilder{}
+}
+
+// SetLogger sets the logger that the handler will use to write to the log. This is mandatory.
+func (b *AlarmMapperBuilder) SetLogger(
+	value *slog.Logger) *AlarmMapperBuilder {
+	b.logger = value
+	return b
+}
+
+// SetBackendClient sets the backend client used to connect to other servers. This is mandatory.
+func (b *AlarmMapperBuilder) SetBackendClient(value *http.Client) *AlarmMapperBuilder {
+	b.backendClient = value
+	return b
+}
+
+// SetResourceServerURL sets the URL of the resource server. This is mandatory.
+// The resource server is used for mapping Alarms to Resources.
+func (b *AlarmMapperBuilder) SetResourceServerURL(
+	value string) *AlarmMapperBuilder {
+	b.resourceServerURL = value
+	return b
+}
+
+// SetResourceServerToken sets the authentication token that will be used to authenticate
+// with to the resource server. This is mandatory.
+func (b *AlarmMapperBuilder) SetResourceServerToken(
+	value string) *AlarmMapperBuilder {
+	b.resourceServerToken = value
+	return b
+}
+
+// SetExtensions sets the fields that will be added to the extensions. This is optional.
+func (b *AlarmMapperBuilder) SetExtensions(values ...string) *AlarmMapperBuilder {
+	b.extensions = values
+	return b
+}
+
+// Build uses the data stored in the builder to create and configure a new handler.
+func (b *AlarmMapperBuilder) Build() (
+	result *AlarmMapper, err error) {
+	// Check parameters:
+	if b.logger == nil {
+		err = errors.New("logger is mandatory")
+		return
+	}
+	if b.backendClient == nil {
+		err = errors.New("backend client is mandatory")
+		return
+	}
```

Comment:
What about the resource server URL and token? Aren't those required?

Reply:
@danielerez can you please answer Juan's question?

Reply:
So a validation for the URL would is indeed needed (as it's mandatory). But the token is optional, as it fallbacks to the backend token if missing. See:
* https://github.com/openshift-kni/oran-o2ims/blob/f9c1a5e7b17c1bafb32f471edb1092e8597fa8c9/internal/service/alarm_handler.go#L148
* https://github.com/openshift-kni/oran-o2ims/blob/ae2a0bb4765e273d06d9befdecf1ae463457716e/internal/cmd/server/start_alarm_server.go#L214